### PR TITLE
Fix queued checkout after a connection failure

### PR DESCRIPTION
**Pull request by charmander**
Attempt at extracting part 1 of #109.

### CODE REVIEW

File: test/error-handling.js [hunk 1]
```diff
@@ -143,4 +144,25 @@ describe('pool error handling', function () {
       return pool.end()
     }))
   })
+
+  it('should continue with queued items after a connection failure', (done) => {
+    const closeServer = net.createServer((socket) => {
+      socket.destroy()
+    }).unref()
```
Comment (brianc) [inline test/error-handling.js]: is the unref needed even though you call close on line 162?  It's not a big deal, just curious.

**Comment by brianc** [comment]
Do you want me to release a new patch version?

**Comment by brianc** [comment]
published `pg-pool@2.0.5`

**Comment by charmander** [comment]
@brianc Thanks!

**Pull request merged**